src/pkg/packager/deploy.go

```diff
@@ -377,19 +377,32 @@ func (p *Packager) pushImagesToRegistry(componentImages []string, noImgChecksum
 
 // Push all of the components git repos to the configured git server
 func (p *Packager) pushReposToRepository(reposPath string, repos []string) error {
-	// Try repo push up to 3 times
 	for _, repoURL := range repos {
-		gitClient := git.New(p.cfg.State.GitServer)
-		repoPath, err := gitClient.TransformURLtoRepoName(repoURL)
-		if err != nil {
-			return fmt.Errorf("unable to get the repo name from the URL %s: %w", repoURL, err)
+
+		// Create an anonymous function to push the repo to the Zarf git server
+		tryPush := func() error {
+			gitClient := git.New(p.cfg.State.GitServer)
+
+			// If this is a serviceURL, create a port-forward tunnel to that resource
+			if tunnel, err := cluster.NewTunnelFromServiceURL(gitClient.Server.Address); err != nil {
```

Comment:
Calling this every time like this breaks external git servers (it works for our "external" test because that is still a service inside the cluster...)

![image](https://user-images.githubusercontent.com/3977569/205704374-fcf35ffd-3b4d-40fd-a8dd-c2d1c036e467.png)

^ tested on https://gitlab.com

Reply:
If you check this pattern first I think it would work for all cases again:

```
	pattern := regexp.MustCompile(`^(?P<name>[^\.]+)\.(?P<namespace>[^\.]+)\.svc\.cluster\.local$`)
```